bin/test-storybook.js

```diff
@@ -34,32 +40,65 @@ function sanitizeURL(url) {
   return finalURL;
 }
 
-const targetURL = sanitizeURL(process.env.TARGET_URL || `http://localhost:6006`);
-
-urlExists(targetURL, function (err, exists) {
-  if (!exists) {
-    console.error(`[test-storybook] It seems that your Storybook instance is not running at: ${targetURL}. Are you sure it's running?`)
-    process.exit(1)
-  }
-
-  executeJestPlaywright()
-});
-
-function executeJestPlaywright() {
-  const fs = require('fs');
-  const path = require('path');
-
+async function executeJestPlaywright(args) {
   const jest = require('jest');
-  let argv = process.argv.slice(2);
+  let argv = args.slice(2);
 
   const jestConfigPath = fs.existsSync('test-runner-jest.config.js')
     ? 'test-runner-jest.config.js'
-    : path.resolve(__dirname, '../playwright/test-runner-jest.config.js')
+    : path.resolve(__dirname, '../playwright/test-runner-jest.config.js');
 
-  argv.push(
-    '--config',
-    jestConfigPath
-  );
+  argv.push('--config', jestConfigPath);
 
-  jest.run(argv);
+  await jest.run(argv);
+}
+
+async function checkStorybook(url) {
+  try {
+    const res = await fetch(url, { method: 'HEAD' });
+    if (res.status !== 200) throw new Error(`Unxpected status: ${res.status}`);
+  } catch (e) {
+    console.error(
+      `[test-storybook] It seems that your Storybook instance is not running at: ${url}. Are you sure it's running?`
+    );
+    process.exit(1);
+  }
 }
+
+async function fetchStoriesJson(url) {
+  const storiesJsonUrl = new URL('/stories.json', url).toString();
+  let tmpFile;
+  try {
+    const res = await fetch(storiesJsonUrl);
+    const tmpDir = tempy.directory();
+    tmpFile = path.join(tmpDir, 'stories-json.test.js');
+    const json = await res.text();
+    const js = transformPlaywrightJson(json);
+    fs.writeFileSync(tmpFile, js);
+  } catch (err) {
+    console.error(`[test-storybook] Failed to fetch stories.json from ${storiesJsonUrl}.`);
```

Comment:
It is possible that the error could be something else here as well. Like stories json is successfully fetched but something else in the code failed, like `transformPlaywrightJson` threw an error that either was supposed to be a bug or an expected error (like Unsupported version), and users would just mention that the error message is "failed to fetch"

Reply:
Also, would be nice to point them to a documentation link in the error message if the error is that the Storybook is running correctly, but just does not support stories.json. The closest doc I found was [this](https://storybook.js.org/docs/react/sharing/storybook-composition#improve-your-storybook-composition) but it doesn't relate with this, we should add to this repo or to Storybook's repo